cmd/ci-operator-prowgen/main.go

```diff
@@ -170,25 +172,46 @@ func getReleaseRepoDir(directory string) (string, error) {
 }
 
 func pruneStaleJobs(jobDir, subDir string) error {
-	if err := jc.OperateOnJobConfigSubdir(jobDir, subDir, func(jobConfig *prowconfig.JobConfig, info *jc.Info) error {
-		pruned := prowgen.Prune(jobConfig)
+	errChan := make(chan error)
+	var errs []error
 
-		if len(pruned.PresubmitsStatic) == 0 && len(pruned.PostsubmitsStatic) == 0 && len(pruned.Periodics) == 0 {
-			if err := os.Remove(info.Filename); err != nil && !os.IsNotExist(err) {
-				return err
-			}
-		} else {
-			if err := jc.WriteToFile(info.Filename, pruned); err != nil {
-				return err
-			}
+	errReadingDone := make(chan struct{})
+	go func() {
```

Comment:
This will run only to close the channel. Instead what we need here is a `for{ select{ ` and grab the case of `errChan` and then we can just append to `errs`. We can also send a `stop` channel to close the goroutin as well.

Reply:
WDYM by "this will run only to close the channel"?

Reply:
Oh, Go seems to wait when it ranges over a channel, so it generally makes sense to design it this way.  Sorry for the noise either way.

Reply:
Not noise, made me recheck at least :) Good thing.